environment.sh

```diff
@@ -36,52 +38,95 @@ done
+git ls-remote --exit-code --heads origin $branch_name >/dev/null 2>&1
+exit_code=$?
+if [[ $exit_code == '0' ]]; then
+  git fetch --all
+  echo "Git branch '$branch_name' exists in the remote repository"
+  git pull -u origin $branch_name
+fi
 
 # Check if environment exists already
-env_exists=$(conda info --envs | grep $env_name | head -n 1)
-if [[ $env_exists == '' ]]; then
-  env_exists="no"
+create_env=$(conda info --envs | grep $env_name | head -n 1)
+if [[ $create_env == '' ]]; then
+  # No environment exists with this name
+  create_env="yes"
 else
-  env_exists="yes"
+  create_env="no"
 fi
 
-if [ $env_exists == 'yes' ]; then
-  creation_time="$(head -n1 /home/$username/miniconda3/envs/$env_name/conda-meta/history)"
+if [ $create_env == 'no' ]; then
+  # Check if existing environment needs to be updated
   echo "Existing environment found for $env_name."
-  # Check if existing environment is older than a week and remake it if so
+  creation_time="$(head -n1 /home/$username/miniconda3/envs/$env_name/conda-meta/history)"
+  # Check if existing environment is older than a week and delete it if so
   if [[ $one_week_ago > $creation_time ]]; then
     echo "Environment is older than one week old. Deleting and remaking environment...."
     conda remove -n $env_name --all -y
-    env_exists="no"
+    create_env="yes"
   fi
+fi
 
-# Create new environment if it did not exist or we need to remake it
-if [ $env_exists == 'no' ]; then
-# This cannot be an else about because we set env_exists to no in that block
+if [ $create_env == 'yes' ]; then
   # Create new environment
   echo "Environment $env_name does not exist. Creating new environment $env_name..."
   # Create conda environment
   conda create -n $env_name python=3.11 -y
-  # Determine which requirements.txt to install from
-  if [ $env_type == 'simulation' ]; then
-    echo "Installing requirements for simulation environment..."
-    install_file="requirements.txt"
-  elif [ $env_type == 'artifact' ]; then
-    echo "Installing requirements for artifact environment"
-    install_file="artifact_requirements.txt"
-  else
-    echo "Invalid environment type. Valid argument types are simulation and artifact."
-    return 
-  fi
 fi
 
-# Activate new environment
+# Activate environment
 conda activate $env_name
+if [ $create_env == 'yes' ]; then
+  # Install json parser
+  conda install jq -y
+fi
+# At this point we have either a new environment or an environment that is less than a week old.
+# We need to check if we need to reinstall packages
+update_env=$create_env
+
+if [ $update_env == 'no' ]; then
+  # Check if environment was build before last modification to requirements file
+  if [[ $creation_time < $requirements_modification_time ]]; then
+    echo "Requirements file last modified after environment was created. Reinstalling packages..."
+    update_env="yes"
+  fi
+  # Check if there has been an update to vivarium packages since last modification to requirements file
```

Comment:
Move this second check outside the outer if statement. Because in the event you've already set `update_env` to yes (due to the requirements.txt being modified after the env was created), you don't need to check the upstream repos - you're going to re-install the env regardless.